File: app/src/main/java/com/cn29/aac/ui/setting/vm/SettingsActivityViewModelFactory.java
package com.cn29.aac.ui.setting.vm;

import javax.inject.Inject;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by Charles Ng on 13/10/2017.
 */

public class SettingsActivityViewModelFactory extends ViewModelProvider.NewInstanceFactory {

  @Inject
  public SettingsActivityViewModelFactory() {
    super();
  }

  @Override
  public <T extends ViewModel> T create(Class<T> modelClass) {
    return (T) new SettingsActivityViewModel();
  }
}

File: app/src/main/java/com/cn29/aac/ui/setting/vm/SettingsActivityViewModelFactory.kt
package com.cn29.aac.ui.setting.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import javax.inject.Inject

/**
 * Created by Charles Ng on 13/10/2017.
 */
class SettingsActivityViewModelFactory @Inject constructor() :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return SettingsActivityViewModel() as T
    }
}